File: Data Structures & Algorithms/house-robber/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums) {
        
        const n = nums.length;
        if(n === 1) return nums[0];

        const dp = [];
        for (let _num in nums) {
            dp.push(-1);
        }

        dp[0] = nums[0];
        dp[1] = Math.max(nums[0], nums[1]);
        for (let i = 2; i < n; i++) {
            dp[i] = Math.max(nums[i] + dp[i - 2], dp[i - 1]);
        }

        return Math.max(dp[n - 1], dp[n - 2]);
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-3.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int n = nums.length;
        int[] result = new int[2 * n];
        int idx = 0;

        for (int j = 0; j < 2; j++) {
            for (int i = 0; i < n; i++) {
                result[idx++] = nums[i];
            }
        }

        return result;
    }
}